### PR TITLE
SLM-83: Allow multiple recipients

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -134,6 +134,8 @@ jobs:
           path: integration_tests/videos
       - store_artifacts:
           path: integration_tests/screenshots
+      - store_artifacts:
+          path: integration_tests/downloads
```
Comment (nathanrussell-moj-digital) [inline .circleci/config.yml:138]: Cypress downloads the PDF file as part of the coversheets journey

File: cypress.json [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
+  "screenshotsFolder": "integration_tests/screenshots",
+  "supportFile": "integration_tests/support/index.ts",
+  "videosFolder": "integration_tests/videos",
+  "reporter": "cypress-multi-reporters",
+  "reporterOptions": {
+    "configFile": "reporter-config.json"
+  },
+  "videoUploadOnPasses": false,
+  "taskTimeout": 60000,
+  "downloadsFolder": "integration_tests/downloads"
```
Comment (nathanrussell-moj-digital) [inline cypress.json:17]: I added this line (to tell cypress where to put the downloads), but for some reason my IDE decided to reformat the rest of the file! 🤷‍♂️

File: integration_tests/pages/barcode/createNewContact.ts [hunk 3]
```diff
@@ -13,21 +13,21 @@ export default class CreateNewContactPage extends Page {
     this.prisonerNameField().should('be.focused')
   }
 
-  submitWithValidValues = (): ReviewRecipientsPage => {
-    this.enterAValidPrisonerName()
-    this.typeAheadAValidPrison()
+  submitWithValidValues = (prisonerName = 'Gage Hewitt', prisonName = 'ashfield'): ReviewRecipientsPage => {
```
Comment (nathanrussell-moj-digital) [inline integration_tests/pages/barcode/createNewContact.ts:16]: Minor refactoring of these methods to give them args with default values. It means the existing tests can call these methods as they do currently (ie. where we dont really care _who_ is added), but for the Legal Sender E2E we want to add several recipients and for them all to be different.

File: server/routes/barcode/contacts/CreateContactController.ts [hunk 4]
```diff
@@ -50,6 +50,8 @@ export default class CreateContactController {
     try {
       const prisonAddress = await this.prisonRegisterService.getPrisonAddress(newRecipient.prisonId)
       this.addRecipient(req, newRecipient, prisonAddress)
+      req.session.findRecipientForm = undefined
```
Comment (nathanrussell-moj-digital) [inline server/routes/barcode/contacts/CreateContactController.ts:53]: After creating each recipient, clear the forms, which allows us to present empty forms to the user when adding more recipients
It also meant I could remove the clearing of the forms in GenerateBarcodeImageController

File: server/views/pages/barcode/review-recipients.njk [hunk 5]
```diff
@@ -21,29 +21,41 @@
...
+
+          {{ govukButton({
+            text: 'Prepare barcodes',
+            href: '/barcode/choose-barcode-option',
+            attributes: { 'data-qa': 'prepare-barcodes-button' }
+          }) }}
+
+        {% else %}
+          <p class="govuk-!-margin-bottom-6">
+            You have removed all recipients. You need to add at least one recipient in order to continue.
```
Comment (nathanrussell-moj-digital) [inline server/views/pages/barcode/review-recipients.njk:50]: Will need some content from the design team here - I'll ask Liora to arrange

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**